pallets/murmur/src/lib.rs

```diff
@@ -131,32 +135,38 @@ pub mod pallet {
 		SchnorrProofVerificationFailed,
 		/// https://crypto.stanford.edu/cs355/19sp/lec5.pdf
 		InvalidSchnorrProof,
-		/// The proxy is not registered as a Murmur wallet or does not exist 
+		/// The proxy is not registered as a Murmur wallet or does not exist
 		InvalidProxy,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Create a time-based proxy account
-		///
+		/// * `name`: The name to assign to the murmur proxy
 		/// * `root`: The MMR root
 		/// * `size`: The size (number of leaves) of the MMR
-		/// * `name`: The name to assign to the murmur proxy
 		/// * `proof`: A (serialized) DLEQ proof
 		/// * `public_key`: A (serialized) public key associated with the DLEQ
 		///
 		#[pallet::weight(0)]
 		#[pallet::call_index(0)]
 		pub fn create(
 			origin: OriginFor<T>,
-			root: BoundedVec<u8, ConstU32<32>>,
+			name: Name,
+			root: Root,
 			size: u64,
-			name: BoundedVec<u8, ConstU32<32>>,
-			proof: BoundedVec<u8, ConstU32<80>>,
-			pubkey: BoundedVec<u8, ConstU32<48>>,
+			proof: Proof,
+			pubkey: SerializedPublicKey,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
+			let nonce = 0;
```

Comment:
Why not go ahead and use a random number for the nonce here?

Reply:
The nonce here is just a monotonic counter to dictate the number of times the wallet has been updated. On creation we set a nonce of zero, then when they update they have to specify a nonce of 1 and so on. It's somewhat analogous to how nonces work in standard wallets, but on a much longer time scale (i.e.  on a 'per session' basis rather than a 'per transaction' basis as usual)